### PR TITLE
Update assumed chain params

### PR DESCRIPTION
**Pull request by maflcko**
> Oh, by the way, the same procedure as last year, Miss Sophie?
> Same procedure as every year, James.

See https://github.com/bitcoin/bitcoin/blob/master/doc/release-process.md#before-branch-off

### CODE REVIEW

**Review by fanquake — APPROVED** [review]
ACK fa0c1b5820d12354d1e31bb561e73165b0adfbac - mainnet, testnet and signet output matches my node.

mainnet blocks & chainstate size:
```bash
du -sh /Users/michael/Library/Application\ Support/Bitcoin/blocks
325G	/Users/michael/Library/Application Support/Bitcoin/blocks
du -sh /Users/michael/Library/Application\ Support/Bitcoin/chainstate 
3.9G	/Users/michael/Library/Application Support/Bitcoin/chainstate
```

notes: https://github.com/fanquake/core-review/blob/master/update-assumevalid.md

File: src/chainparams.cpp [hunk 1]
```diff
@@ -105,8 +105,8 @@ class CMainParams : public CChainParams {
         pchMessageStart[3] = 0xd9;
         nDefaultPort = 8333;
         nPruneAfterHeight = 100000;
-        m_assumed_blockchain_size = 320;
-        m_assumed_chain_state_size = 4;
+        m_assumed_blockchain_size = 350;
+        m_assumed_chain_state_size = 6;
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/chainparams.cpp:109]: Mine is almost 5 :man_shrugging: 


```
$ du -sh ./chainstate/
4.8G	./chainstate/

[the remark this answers is not in the document]
Comment (darosior) [reply src/chainparams.cpp:109]: Mine is 4, fwiw.

**Review by darosior — APPROVED** [review]
ACK fa0c1b5820d12354d1e31bb561e73165b0adfbac mainnet and testnet params.

File: src/chainparams.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonatack) [reply src/chainparams.cpp:109]: 3.9 here like @fanquake

File: src/chainparams.cpp [hunk 2]
```diff
@@ -162,10 +162,10 @@ class CMainParams : public CChainParams {
...
 
         chainTxData = ChainTxData{
-            // Data from RPC: getchaintxstats 4096 0000000000000000000f2adce67e49b0b6bdeb9de8b7c3d7e93b21e7fc1e819d
-            /* nTime    */ 1585764811,
-            /* nTxCount */ 517186863,
-            /* dTxRate  */ 3.305709665792344,
+            // Data from RPC: getchaintxstats 4096 0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72
+            /* nTime    */ 1603995752,
+            /* nTxCount */ 582083445,
+            /* dTxRate  */ 3.508976121410527,
```
Comment (jonatack) [inline src/chainparams.cpp:168]: Mainnet chaintxdata seem good:
```
$ ./src/bitcoin-cli getchaintxstats 4096 0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72
{
  "time": 1603995752,
  "txcount": 582083445,
  "window_final_block_hash": "0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72",
  "window_final_block_height": 654683,
  "window_block_count": 4096,
  "window_tx_count": 8995148,
  "window_interval": 2563468,
  "txrate": 3.508976121410527
}
```

File: src/chainparams.cpp [hunk 3]
```diff
@@ -251,10 +251,10 @@ class CTestNetParams : public CChainParams {
...
 
         chainTxData = ChainTxData{
-            // Data from RPC: getchaintxstats 4096 000000000000056c49030c174179b52a928c870e6e8a822c75973b7970cfbd01
-            /* nTime    */ 1585561140,
-            /* nTxCount */ 13483,
-            /* dTxRate  */ 0.08523187013249722,
+            // Data from RPC: getchaintxstats 4096 000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0
+            /* nTime    */ 1603359686,
+            /* nTxCount */ 58090238,
+            /* dTxRate  */ 0.1232886622799463,
```
Comment (jonatack) [inline src/chainparams.cpp:257]: Testnet3 chaintxdata seem good:
```
$ ./src/bitcoin-cli -testnet getchaintxstats 4096 000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0
{
  "time": 1603359686,
  "txcount": 58090238,
  "window_final_block_hash": "000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0",
  "window_final_block_height": 1864000,
  "window_block_count": 4096,
  "window_tx_count": 143148,
  "window_interval": 1161080,
  "txrate": 0.1232886622799463
}
```

File: src/chainparams.cpp [hunk 4]
```diff
@@ -274,15 +274,15 @@ class SigNetParams : public CChainParams {
...
             m_assumed_chain_state_size = 0;
             chainTxData = ChainTxData{
                 // Data from RPC: getchaintxstats 4096 00000128807d9175c494e24d805fc7854f7d79aa965cbb128342ad8b70cecfa5
-                /* nTime    */ 1601382000,
-                /* nTxCount */ 5435,
-                /* dTxRate  */ 0.001898346323372538,
+                /* nTime    */ 1603986000,
+                /* nTxCount */ 9582,
+                /* dTxRate  */ 0.00159272030651341,
             };
```
Comment (jonatack) [inline src/chainparams.cpp:286]: I have different Signet chaintxdata:
```
$ ./src/bitcoin-cli -signet getchaintxstats 4096 00000128807d9175c494e24d805fc7854f7d79aa965cbb128342ad8b70cecfa5
{
  "time": 1601382000,
  "txcount": 5435,
  "window_final_block_hash": "00000128807d9175c494e24d805fc7854f7d79aa965cbb128342ad8b70cecfa5",
  "window_final_block_height": 5348,
  "window_block_count": 4096,
  "window_tx_count": 4182,
  "window_interval": 2202970,
  "txrate": 0.001898346323372538
}
```

**Review by jonatack — COMMENTED** [review]
On mainnet, I see a blockchain size of 326.2 and a chainstate size of 3.9. 

The chaintxdata for me are the same as this PR, except for signet.

File: src/chainparams.cpp (hunk 4 above)
Comment (jonatack) [reply src/chainparams.cpp:286]: Maybe I'm on a previous version of signet?

Comment (jonatack) [reply src/chainparams.cpp:286]: Seeing the same output on a fresh new synced signet chain.

File: src/chainparams.cpp [hunk 5]
```diff
@@ -91,8 +91,8 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nStartTime = 1199145601; // January 1, 2008
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nTimeout = 1230767999; // December 31, 2008
 
-        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000e1ab5ec9348e9f4b8eb8154");
-        consensus.defaultAssumeValid = uint256S("0x0000000000000000000f2adce67e49b0b6bdeb9de8b7c3d7e93b21e7fc1e819d"); // 623950
+        consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000001533efd8d716a517fe2c5008");
+        consensus.defaultAssumeValid = uint256S("0x0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72"); // 654683
```
Comment (jonatack) [inline src/chainparams.cpp:95]: The mainnet nMinimumChainWork and defaultAssumeValid consensus params match up for me
 ```
$ ./src/bitcoin-cli getblockheader 0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72
{
  "hash": "0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72",
  "height": 654683,
  "chainwork": "00000000000000000000000000000000000000001533efd8d716a517fe2c5008",
}
````

File: src/chainparams.cpp [hunk 6]
```diff
@@ -204,8 +204,8 @@ class CTestNetParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nStartTime = 1199145601; // January 1, 2008
         consensus.vDeployments[Consensus::DEPLOYMENT_TAPROOT].nTimeout = 1230767999; // December 31, 2008
 
-        consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000000001495c1d5a01e2af8a23");
-        consensus.defaultAssumeValid = uint256S("0x000000000000056c49030c174179b52a928c870e6e8a822c75973b7970cfbd01"); // 1692000
+        consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000000001db6ec4ac88cf2272c6");
+        consensus.defaultAssumeValid = uint256S("0x000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0"); // 1864000
```
Comment (jonatack) [inline src/chainparams.cpp:208]: Testnet nMinimumChainWork and defaultAssumeValid consensus params match up for me
```
$ ./src/bitcoin-cli -testnet getblockheader 000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0
{
  "hash": "000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0",
  "height": 1864000,
  "chainwork": "0000000000000000000000000000000000000000000001db6ec4ac88cf2272c6",
}
```

**Review by jonatack — COMMENTED** [review]
ACK for mainnet and testnet, not sure why I have different data for signet.

File: src/chainparams.cpp [hunk 7]
```diff
@@ -274,15 +274,15 @@ class SigNetParams : public CChainParams {
...
             vSeeds.emplace_back("ntv3mtqw5wt63red.onion:38333");
 
-            consensus.nMinimumChainWork = uint256S("0x0000000000000000000000000000000000000000000000000000000d145533ce");
-            consensus.defaultAssumeValid = uint256S("0x00000128807d9175c494e24d805fc7854f7d79aa965cbb128342ad8b70cecfa5"); // 5348
+            consensus.nMinimumChainWork = uint256S("0x00000000000000000000000000000000000000000000000000000019fd16269a");
+            consensus.defaultAssumeValid = uint256S("0x0000002a1de0f46379358c1fd09906f7ac59adf3712323ed90eb59e4c183c020"); // 9434
             m_assumed_blockchain_size = 1;
             m_assumed_chain_state_size = 0;
             chainTxData = ChainTxData{
                 // Data from RPC: getchaintxstats 4096 00000128807d9175c494e24d805fc7854f7d79aa965cbb128342ad8b70cecfa5
```
Comment (MarcoFalke) [inline src/chainparams.cpp:282]:
```suggestion
                // Data from RPC: getchaintxstats 4096 0000002a1de0f46379358c1fd09906f7ac59adf3712323ed90eb59e4c183c020
```

The signet comment was not updated

File: src/chainparams.cpp (hunk 4 above)
Comment (MarcoFalke) [reply src/chainparams.cpp:286]: The comment has not been updated https://github.com/bitcoin/bitcoin/pull/20263/files#r515620099 . Nice find.

**Review by theStack — COMMENTED** [review]
Verified mainnet, testnet3, signet params via the following commands (starting with block height as first input feels most natural to me):
```
$ bitcoin-cli getblockheader $(bitcoin-cli getblockhash 654683) | grep "\"chainwork\|\"hash"
$ bitcoin-cli getchaintxstats 4096 0000000000000000000b9d2ec5a352ecba0592946514a92f14319dc2b367fc72

$ bitcoin-cli -testnet getblockheader $(bitcoin-cli -testnet getblockhash 1864000) | grep "\"chainwork\|\"hash"
$ bitcoin-cli -testnet getchaintxstats 4096 000000000000006433d1efec504c53ca332b64963c425395515b01977bd7b3b0

$ bitcoin-cli -signet getblockheader $(bitcoin-cli -signet getblockhash 9434) | grep "\"chainwork\|\"hash"
$ bitcoin-cli -signet getchaintxstats 4096 0000002a1de0f46379358c1fd09906f7ac59adf3712323ed90eb59e4c183c020
```

ACK (modulo the `getchaintxstats` signet comment that has not been updated yet (https://github.com/bitcoin/bitcoin/pull/20263#discussion_r515620099).

File: src/chainparams.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (theStack) [reply src/chainparams.cpp:109]: Having 4.0G here for chainstate.

[the remark this answers is not in the document]
Comment (sipa) [reply src/chainparams.cpp:109]: $ du -sh .bitcoin/chainstate/
4.0G    .bitcoin/chainstate/

**Comment by MarcoFalke** [comment]
I force pushed to fixup the signet comment. Should be easy to re-ACK for those that already ACKed

**Review by dergoegge — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/commit/fa90ba36d3f640ef2d559b69ae1ea99a73da6ac8 - mainnet and testnet data matches my node.

Followed process as outlined here: https://github.com/fanquake/core-review/blob/master/update-assumevalid.md

File: src/chainparams.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/chainparams.cpp:109]: Mine is 5.2G

**Comment by jonatack** [comment]
ACK fa90ba36d3f640ef2d559b69ae1ea99a73da6ac8 per `git diff fa0c1b5 fa90ba3` and re-running getblockheader/getblockhash/getchaintxstats on the signet chain

**Review by theStack — APPROVED** [review]
re-ACK fa90ba36d3f640ef2d559b69ae1ea99a73da6ac8 ✔️

**Review by darosior — APPROVED** [review]
re-ACK fa90ba36d3f640ef2d559b69ae1ea99a73da6ac8 for mainnet and testnet.

**Pull request merged**